src/plots/plots.js

```diff
@@ -505,12 +506,38 @@ plots.supplyDefaults = function(gd) {
     // update object references in calcdata
     if((gd.calcdata || []).length === newFullData.length) {
```

Comment:
This block update the `gd.calcdata[i][0].trace` reference so that the plot step can use the correct `fullData` attribute values even in cases where `Plots.doCalcdata` isn't called (e.g. on zoom).